src/simularium/SelectionInterface.ts

```diff
@@ -234,6 +246,70 @@ class SelectionInterface {
             };
         });
     }
+
+    public setAgentColors(
+        uiDisplayData: UIDisplayData,
+        colors: (string | number)[],
+        visGeometry: VisGeometry | VisGeometryMock
```

Comment:
yeah I agree with this last thing, but I've struggle with how to separate it since the visgeometry is the one holding and managing the colors, but the selection interface seems like the logical place to handle this looking forward when users can change the color of an agent through the front end. I'm going to push some changes that address the multiple calls to createMaterials, and then I'll take another look at separating this out.